network/tcp_client_test.cc

```diff
@@ -0,0 +1,267 @@
+}
+
+TEST_F(TcpClientFakeTest, DownloadFileFromServerSuccess)
+{
+    std::string file_content = "Mock file content data";
+    std::string local_path = (std::filesystem::temp_directory_path() / "test_download.tmp")
+                             .string();
+
+    // Configure server to send response AND file data immediately
+    m_fake_server.SetHandler(
+    [file_content](FakeSocketConnection* conn, std::unique_ptr<ISerializable> req) {
+        ASSERT_EQ(req->GetMessageType(), MessageType::DOWNLOAD_FILE_REQUEST);
+
+        DownloadFileResponse resp;
+        resp.SetFound(true);
+        resp.SetFileSizeStr(std::to_string(file_content.size()));
+        ASSERT_TRUE(SendSocketMessage(conn, resp).ok());
+
+        // Immediately send the "file" raw data through the fake connection
+        ASSERT_TRUE(
+        conn->Send(reinterpret_cast<const uint8_t*>(file_content.data()), file_content.size())
+        .ok());
+    });
+
+    auto status = m_client->DownloadFileFromServer("/remote/file", local_path);
+    ASSERT_TRUE(status.ok()) << status.message();
+
+    // Verify file was written to disk by TcpClient's base SocketConnection::ReceiveFile
+    std::ifstream ifs(local_path, std::ios::binary);
+    ASSERT_TRUE(ifs.good());
+    std::string read_content((std::istreambuf_iterator<char>(ifs)),
+                             std::istreambuf_iterator<char>());
+    EXPECT_EQ(read_content, file_content);
+
+    ifs.close();
+    std::filesystem::remove(local_path);
+}
+
+TEST_F(TcpClientFakeTest, DownloadFileNotFound)
+{
+    m_fake_server.SetHandler([](FakeSocketConnection* conn, std::unique_ptr<ISerializable> req) {
+        DownloadFileResponse resp;
+        resp.SetFound(false);
+        resp.SetErrorReason("Restricted access");
+        ASSERT_TRUE(SendSocketMessage(conn, resp).ok());
+    });
+
+    auto status = m_client->DownloadFileFromServer("/remote/secret", "dont_care.tmp");
+    EXPECT_FALSE(status.ok());
+    EXPECT_EQ(status.code(), absl::StatusCode::kNotFound);
+    EXPECT_TRUE(std::string(status.message()).find("Restricted access") != std::string::npos);
+}
+
+TEST_F(TcpClientFakeTest, StartPm4CaptureSuccess)
+{
+    m_fake_server.SetHandler([](FakeSocketConnection* conn, std::unique_ptr<ISerializable> req) {
+        ASSERT_EQ(req->GetMessageType(), MessageType::PM4_CAPTURE_REQUEST);
+        Pm4CaptureResponse resp;
+        resp.SetString("/var/log/capture.pm4");
+        ASSERT_TRUE(SendSocketMessage(conn, resp).ok());
+    });
+
+    auto result = m_client->StartPm4Capture();
+    ASSERT_TRUE(result.ok());
+    EXPECT_EQ(*result, "/var/log/capture.pm4");
+}
+
+TEST_F(TcpClientFakeTest, DisconnectStopsKeepAlive)
+{
+    EXPECT_TRUE(m_client->IsConnected());
+    m_client->Disconnect();
+    EXPECT_FALSE(m_client->IsConnected());
+    // Wait a bit to ensure no crashes from background threads
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Please don't sleep in tests. Try to replace the sleep with a more explicit wait.

If necessary, prefer to use abseil time library
```suggestion
    absl::SleepFor(absl::Milliseconds(100));
```